packages/flutter_tools/test/integration.shard/debug_adapter/flutter_adapter_test.dart

```diff
@@ -354,6 +354,36 @@ void main() {
 
       await dap.client.terminate();
     });
+
+   testWithoutContext('can attach to an already-running Flutter app and detach', () async {
+      final Uri vmServiceUri = await testProcess.vmServiceUri;
+      bool debugeeTerminated = false;
+      unawaited(testProcess.process.exitCode.then((_) => debugeeTerminated = true));
+
+      // Launch the app and wait for it to print "topLevelFunction".
+      await Future.wait(<Future<void>>[
+        dap.client.stdoutOutput.firstWhere(
+            (String output) => output.startsWith('topLevelFunction')),
+        dap.client.start(
+          launch: () => dap.client.attach(
+            cwd: project.dir.path,
+            toolArgs: <String>['-d', 'flutter-tester'],
+            vmServiceUri: vmServiceUri.toString(),
+          ),
+        ),
+      ], eagerError: true);
+
+      // Send a terminate request which should terminate the debugger, but not
+      // terminate the debugee
+      await Future.wait(<Future<void>>[
+        dap.client.event('terminated'),
+        dap.client.terminate(),
+      ]);
+
+      // Wait a short amount before checking to ensure it really didn't quit.
```

Comment:
Does this test flake without the timeout?

Reply:
It doesn't flake, but the test seems less valid. We're trying to prove the process _did not_ quit, so if we check too fast, it's possible we think it didn't quit, but it just didn't quit _yet_.

If I remove the timeout, the test passes even without the fix, but with the timeout it fails without the fix and passes with the fix.

It doesn't feel like a perfect solution, but I don't know there's another good way to ensure it didn't happen for a period.

Reply:
Yeah, that makes sense. I always struggle with how to write these types of tests. I will start a thread in the discord....